### PR TITLE
impt(test): add tests for constant exchange v 1

### PR DESCRIPTION
**Pull request by mattwasielewski**

### CODE REVIEW

File: src/pool.spec.ts [hunk 1]
```diff
@@ -215,20 +216,43 @@ describe("Generic pool", () => {
...
-    expect(pool.primaryAsset.index).toBe(testBed.algo.index);
-    expect(pool.secondaryAsset.index).toBe(testBed.coin.index);
+  expect(pool.internalState).toEqual(state);
+}
 
-    expect(pool.poolType).toBe("CONSTANT_PRODUCT");
-    expect(pool.version).toBe(2);
+describe("Constant product pool", () => {
+  it("parsing state version 1", async () => {
+    const pact = new PactClient(algod);
```
Comment (tulustul) [inline src/pool.spec.ts:236]: `pact` is inside `testbed`

File: src/pool.spec.ts [hunk 2]
```diff
@@ -215,20 +216,43 @@ describe("Generic pool", () => {
...
+    const testBed = await makeFreshTestBed({ poolType: "CONSTANT_PRODUCT" }, 1);
+    const pool = await pact.fetchPoolById(testBed.pool.appId);
```
Comment (tulustul) [inline src/pool.spec.ts:238]: just use `testBed.pool`

File: src/testUtils.ts [hunk 3]
```diff
@@ -62,13 +62,15 @@ export function deployExchangeContract(
   options: {
     feeBps?: number;
   } = {},
+  version?: number,
```
Comment (tulustul) [inline src/testUtils.ts:65]: put it into `options`

File: src/testUtils.ts [hunk 4]
```diff
@@ -101,23 +105,38 @@ export function deployContract(
...
-   --amplifier=${(options.amplifier ?? 80) * 1000} \\
-   --admin_and_treasury_address=${account.addr}
-   `;
+    --contract-type=${poolType.toLowerCase()} \\
+    --primary_asset_id=${primaryAssetIndex} \\
+    --secondary_asset_id=${secondaryAssetIndex} \\
+    --fee_bps=${options.feeBps ?? 30} \\
+    --pact_fee_bps=${options.pactFeeBps ?? 0} \\
+    --amplifier=${(options.amplifier ?? 80) * 1000} \\
+    --admin_and_treasury_address=${account.addr}
```
Comment (tulustul) [inline src/testUtils.ts:138]: Can we avoid the repetitions here?

File: src/testUtils.ts [hunk 5]
```diff
@@ -203,16 +223,23 @@ export async function makeFreshTestBed(
...
-    poolType,
-    algo.index,
-    coin.index,
-    {
-      feeBps: options.feeBps,
-      amplifier: options.amplifier,
-    },
-  );
+
+  const appId = version
```
Comment (tulustul) [inline src/testUtils.ts:227]: No need for ternary operator, just pass null/undefined as a version

File: src/pool.spec.ts [hunk 6]
```diff
@@ -400,6 +439,23 @@ describe("Constant product pool", () => {
     await signAndSend(removeLiqTxGroup, account);
     await pool.updateState();
     expect(pool.state.totalLiquidity).toBe(1000);
+  }
+
+  it("Pool e2e scenario for asset with 19 decimals", async () => {
+    const testBed = await makeFreshTestBed({
+      poolType: "CONSTANT_PRODUCT",
+    });
+    e2e_scenario_19_decimals(testBed);
```
Comment (tulustul) [inline src/pool.spec.ts:448]: No need to test `19_decimals` variant for old contracts.

**Review by tulustul — CHANGES_REQUESTED** [review]

File: src/pool.spec.ts (hunk 1 above)
Comment (mattwasielewski) [reply src/pool.spec.ts:236]: DONE

File: src/pool.spec.ts (hunk 2 above)
Comment (mattwasielewski) [reply src/pool.spec.ts:238]: DONE

File: src/testUtils.ts (hunk 3 above)
Comment (mattwasielewski) [reply src/testUtils.ts:65]: DONE

File: src/testUtils.ts (hunk 4 above)
Comment (mattwasielewski) [reply src/testUtils.ts]: DONE

File: src/testUtils.ts (hunk 5 above)
Comment (mattwasielewski) [reply src/testUtils.ts]: DONE

File: src/pool.spec.ts (hunk 6 above)
Comment (mattwasielewski) [reply src/pool.spec.ts]: DONE

**Review by tulustul — APPROVED** [review]

**Pull request merged**